cloud/amazon/lambda.py

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+DOCUMENTATION = '''
+---
+module: lambda
+short_description: Manage AWS Lambda functions
+description:
+     - Allows for the management of Lambda functions.
+version_added: '2.2'
+requirements: [ boto3, base64, hashlib ]
+options:
+  name:
+    description:
+      - The name you want to assign to the function you are uploading. Cannot be changed.
+    required: true
+  state:
+    description:
+      - Create or delete Lambda function
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  runtime:
+    description:
+      - The runtime environment for the Lambda function you are uploading. Required when creating a function.
+    default: null
+    choices: [ 'nodejs', 'nodejs4.3', 'java8', 'python2.7' ]
+  role_arn:
+    description:
+      - The Amazon Resource Name (ARN) of the IAM role that Lambda assumes when it executes your function to access any other Amazon Web Services (AWS) resources
+    default: null
+  handler:
+    description:
+      - The function within your code that Lambda calls to begin execution
+    default: null
+  path:
```

Comment:
Can this be renamed `zip_path` or similar, so the expected format is part of the parameter name?